Replace reference to deprecated exception class

colemanw opened the pull request:
The api-specific exception classes have been deprecated and will soon be removed from core.
They are identical to CRM_Core_Exception.

colemanw committed:
Replace reference to deprecated exception class

The api-specific exception classes have been deprecated and will soon be removed from core.
They are identical to CRM_Core_Exception.

twomice commented:
@colemanw Actually, none of these functions throws any exception. "Old me" apparently thought `@throws` was warranted because "down in there somewhere" one could be thrown, but "now me" thinks that's just silly. I think the right solution here is just to remove all 3 of these lines.

colemanw commented:
@twomice I think the basic api3 functions being delegated to do throw `CRM_Core_Exceptions`... but removing the lines is also fine!

twomice committed:
Remove needless @throws tags

My current position is that @throws is only appropriate for functions that contain the keyword `throw`. Otherwise we'd need this tag for almost any function in a civicrm extension.

twomice commented:
Done. My current position is that `@throws` is only appropriate for functions that contain the keyword `throw`. Otherwise we'd need this tag for almost any function in a civicrm extension.

colemanw commented:
Fair enough!
